sdk/sync/write.go

```diff
@@ -50,30 +50,34 @@ func NewWriter(runenv *runtime.RunEnv) (w *Writer, err error) {
 		refreshset: make(map[string]struct{}),
 	}
 
-	go w.refreshOwned()
+	go w.refreshWorker()
 	return w, nil
 }
 
-// refreshOwned runs a loop that refreshes owned keys every `RefreshPeriod`.
+// refreshWorker runs a loop that refreshes owned keys every `RefreshPeriod`.
 // It should be launched as a goroutine.
-func (w *Writer) refreshOwned() {
-Loop:
-	select {
-	case <-time.After(RefreshPeriod):
-		w.lk.RLock()
-		// TODO: do this in a transaction. We risk the loop overlapping with the
-		// refresh period, and all kinds of races. We need to be adaptive here.
-		for k, _ := range w.refreshset {
-			if err := w.client.Expire(k, TTL).Err(); err != nil {
-				w.lk.RUnlock()
-				panic(err)
-			}
+func (w *Writer) refreshWorker() {
+	for {
+		select {
+		case <-time.After(RefreshPeriod):
+			w.refreshOwned()
+		case <-w.doneCh:
+			return
 		}
-		w.lk.RUnlock()
-		goto Loop
+	}
+}
 
-	case <-w.doneCh:
-		return
+// refreshOwned refreshes all owned keys.
+func (w *Writer) refreshOwned() {
```

Comment:
Can I kindly ask for an improved explanation? Reading through this function the first time, it seems to me that it is checking which nodes are still alive and sets a TTL for their key. However, my understanding either doesn't seem complete or right. Can I haz some wisdom sprinkles?

Reply:
Yeah, let me enhance the comments in that source file!